src/interpreter.rs

```diff
@@ -39,8 +43,8 @@ struct Word<T> {
 }
 
 impl<T> Word<T> {
-    fn eval_within(&self, i: &mut Interpreter<T>) {
-        (self.entry)(i)
+    fn eval_within(&self, mut i: &mut Interpreter<T>) {
+      (self.entry)(&mut i)
     }
```

Comment:
We pass a mutabl reference to the interpreter. There is no ownership loop here, since we only manipulate a reference